lib/persistence/repositories/impl/_AccountRepository.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:coffeecard/model/Login.dart';
+import 'package:coffeecard/persistence/http/RestClient.dart';
+import 'package:coffeecard/persistence/repositories/AccountRepository.dart';
+import 'package:coffeecard/persistence/storage/SecureStorage.dart';
+import 'package:dio/dio.dart';
+import 'package:logger/logger.dart';
+
+class _AccountRepository implements AccountRepository {
+  final RestClient _restClient;
+  final Logger _logger;
+  final SecureStorage _storage;
+
+  _AccountRepository(this._restClient, this._logger, this._storage);
+
+  @override
+  Future login(String userName, String password) async {
+    var login = Login(userName, password, "");
+
+    var token = await _restClient.login(login).catchError((Object obj) {
+      switch (obj.runtimeType) {
```

Comment:
Any reason why we are not using a standard try - catch for error handling?

Reply:
I honestly don't see where on that page it says this is the recommended way to do it. It even has a pretty large disclaimer saying:

"You don’t need this page if your code uses the language’s asynchrony support — async, await, and error handling using try-catch. For more information, see the asynchronous programming codelab."

Is it because retrofit is "old fashioned" and reliant on using future methods?

Reply:
Good found with the disclaimer. I guess it's a taste which way you like. In this case you're right that the try/catch block makes more sense. I'll rewrite

Reply:
Good to hear, I take it you will be rewriting the appconfig one as well then?

Reply:
Done. See latest commit